test/e2e/e2e_toomanyrestarts_test.go

```diff
@@ -0,0 +1,190 @@
+
+	nodeList, err := clientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		t.Errorf("Error listing node with %v", err)
+	}
+
+	nodes, workerNodes := splitNodesAndWorkerNodes(nodeList.Items)
+
+	t.Log("Creating testing namespace")
+	testNamespace := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "e2e-" + strings.ToLower(t.Name())}}
+	if _, err := clientSet.CoreV1().Namespaces().Create(ctx, testNamespace, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Unable to create ns %v", testNamespace.Name)
+	}
+	defer clientSet.CoreV1().Namespaces().Delete(ctx, testNamespace.Name, metav1.DeleteOptions{})
+
+	t.Log("Creating pods with different restartCount")
+	for i := int32(0); i < 4; i++ {
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("restart-pod-%v", i),
+				Namespace: testNamespace.Name,
+				Labels:    map[string]string{"test": "restart-pod", "name": "test-toomanyrestarts"},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{{
+					Name:            "pause",
+					ImagePullPolicy: "Never",
+					Image:           "kubernetes/pause",
+					Ports:           []v1.ContainerPort{{ContainerPort: 80}},
+				}},
+			},
+		}
+		pod.Status = v1.PodStatus{
+			InitContainerStatuses: []v1.ContainerStatus{
+				{
+					RestartCount: 5 * i,
+				},
+			},
+			ContainerStatuses: []v1.ContainerStatus{
+				{
+					RestartCount: 10 * i,
+				},
+				{
+					RestartCount: 10 * i,
+				},
+			},
+		}
+
+		t.Logf("Creating pod %v in %v namespace for node %v", pod.Name, pod.Namespace, workerNodes[0].Name)
+		_, err := clientSet.CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{})
+		if err != nil {
+			t.Logf("Error creating restart pods: %v", err)
+			if err = clientSet.CoreV1().Pods(pod.Namespace).DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{
+				LabelSelector: labels.SelectorFromSet(labels.Set(map[string]string{"test": "restart-pod", "name": "toomanyrestarts"})).String(),
+			}); err != nil {
+				t.Fatalf("Unable to delete restart pods: %v", err)
+			}
+			return
+		}
+	}
+
+	tests := []struct {
+		name                    string
+		podRestartThreshold     int32
+		includingInitContainers bool
+		expectedEvictedPodCount int
+	}{
+		{
+			name:                    "test-no-evictions",
+			podRestartThreshold:     int32(10000),
+			includingInitContainers: true,
+			expectedEvictedPodCount: 0,
+		},
+		{
+			name:                    "test-three-evictions",
+			podRestartThreshold:     int32(1),
+			includingInitContainers: false,
+			expectedEvictedPodCount: 3,
+		},
+		{
+			name:                    "test-one-evictions",
+			podRestartThreshold:     int32(70),
+			includingInitContainers: true,
+			expectedEvictedPodCount: 1,
+		},
+	}
+	for _, tc := range tests {
+		t.Log("Creating RC with 4 replicas owning the created pods")
+		rc := RcByNameContainer("toomanyrestarts", testNamespace.Name, int32(4), map[string]string{"test": "restart-pod"}, nil, "")
```

Comment:
`RcByNameContainer` creates a pod template without the pod.spec set. So the test must fail in any case. Also, I don't think `clientSet.CoreV1().Pods(pod.Namespace).Create` will create a pod with the pod.Status as provided (maybe I am wrong?). One needs to use `Pods(...).Status(...).Update subresource instead. 

Can you move `t.Log("Creating pods with different restartCount")` part into `for _, tc := range tests` and have the `RestartCount` as part of the test struct? Then creating the pods with the `RestartCount` in every test iteration right before `RcByNameContainer` is invoked?

Reply:
Yes, I will fix, much thanks :)

Reply:
I try to use `if _, err := clientSet.CoreV1().Pods(pod.Namespace).UpdateStatus(ctx, pod, metav1.UpdateOptions{}); err != nil {` to update the restartCount, it success, but very soon, about 2 seconds, the kubelet will rollback the restartCount back to 0.
So I change to only test duplicatePods.

Reply:
The kubelet will not let himself to be fooled quite quickly. You might create pods that will constantly crash (e.g. running `sleep 1s && exit 1`).

Reply:
very good idea~ Before your comment, I just use `kubectl exec -it %s -n %s -- /bin/sh -c "kill 1"` for kill pod :( 
I will fix soon~ Thanks a lot~